packages/flutter_app_badger/tizen/src/tizen_badge.cc

```diff
@@ -0,0 +1,96 @@
+// Copyright 2022 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "tizen_badge.h"
+
+#include "log.h"
+
+bool TizenBadge::Initialize() {
+  char *app_id = nullptr;
+  int ret = app_get_id(&app_id);
+  if (ret != APP_ERROR_NONE) {
+    LOG_ERROR("app_get_id() failed with error %d.", ret);
+    return false;
+  }
+  app_id_ = app_id;
+  free(app_id);
+
+  ret = badge_add(app_id_.c_str());
+  if (ret == BADGE_ERROR_NOT_SUPPORTED) {
+    LOG_DEBUG("Badge is not supported.");
+    is_supported_ = false;
+  } else if (ret != BADGE_ERROR_NONE && ret != BADGE_ERROR_ALREADY_EXIST) {
+    LOG_ERROR("badge_add() failed with error %d.", ret);
+    return false;
+  } else {
+    is_supported_ = true;
+  }
+
+  is_added_ = true;
+  return true;
+}
+
+bool TizenBadge::UpdateBadgeCount(int count) {
+  if (!is_supported_) {
+    LOG_ERROR("Badge is not supported.");
+    last_error_ = BADGE_ERROR_NOT_SUPPORTED;
+    return false;
+  }
+  int ret = BADGE_ERROR_NONE;
+  if (!is_added_) {
+    ret = badge_add(app_id_.c_str());
+    if (ret != BADGE_ERROR_NONE && ret != BADGE_ERROR_ALREADY_EXIST) {
+      LOG_ERROR("badge_add() failed with error %d.", ret);
+      last_error_ = ret;
+      return false;
+    }
+    is_added_ = true;
+  }
+
+  ret = badge_set_count(app_id_.c_str(), count);
+  if (ret != BADGE_ERROR_NONE) {
+    LOG_ERROR("badge_set_count() failed with error %d.", ret);
+    last_error_ = ret;
+    return false;
+  }
+
+  ret = badge_set_display(app_id_.c_str(), 1);
+  if (ret != BADGE_ERROR_NONE) {
+    LOG_ERROR("badge_set_display() failed with error %d.", ret);
+    last_error_ = ret;
+    return false;
+  }
+  return true;
+}
+
+bool TizenBadge::RemoveBadge() {
+  if (!is_supported_) {
+    LOG_ERROR("Badge is not supported.");
+    last_error_ = BADGE_ERROR_NOT_SUPPORTED;
+    return false;
+  }
+
+  int ret = badge_set_display(app_id_.c_str(), 0);
```

Comment:
Do you know how `badge_remove` is different from setting `badge_set_display` to 0?

Reply:
As far as I know badge_remove removes the internally created badge object from badge_add.
(If we don't do badge_remove, ALREADY_EXSIT will be raised when we do badge_add.)
and I checked the sample code related to the badge api here.
https://docs.tizen.org/development/sample/native/AppFW/Badges/src/main.c

Reply:
I thought then it would be enough to call only `badge_remove` in `RemoveBadge` but it's up to you.

Reply:
i removed badge_set_count() and badge_remove() calls. I haven't found a more good reason to keep them.